Algorithms.Tests/Search/FastSearcherTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using Algorithms.Search;
+
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Algorithms.Tests.Search
+{
+    public static class FastSearcherTests
+    {
+
+        [Test]
+        public static void FindIndex_ItemPresent_IndexCorrect()
+        {
+            // Arrange
+            var searcher = new FastSearcher();
+            var arr = new int[10];
+            for (var i = 0; i < arr.Length; i++)
+            {
+                arr[i] = i;
+            }
+
+            Array.Sort(arr);
```

Comment:
What for? Isn't it sorted already?

Reply:
Of course it is, but one of the dependencies of the program is that the array must be sorted and I thought it would be easier to see it that way. I will remove it because it's redundant.

Reply:
I see your point here, I think you can even leave it if you wish as it makes sense there as an educational point, even though it's redundant. Just add a comment that explains it.